src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalDirectoryInfo.cs

```diff
@@ -55,5 +67,40 @@ public Stream CreateReadStream()
         {
             throw new InvalidOperationException(SR.CannotCreateStream);
         }
+
+        /// <inheritdoc/>
+        public IEnumerator<IFileInfo> GetEnumerator()
+        {
+            EnsureInitialized();
+            return _entries.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            EnsureInitialized();
+            return _entries.GetEnumerator();
+        }
+
+        [MemberNotNull(nameof(_entries))]
+        private void EnsureInitialized()
+        {
+            try
+            {
+                _entries = _info
```

Comment:
Not related to this change but what's the point of `_entries` if we assign it every time it will be used?